eras/shelley/impl/src/Cardano/Ledger/Shelley/Genesis.hs

```diff
@@ -167,6 +167,43 @@ instance Era era => ToJSON (ShelleyGenesis era) where
         "initialFunds" .= sgInitialFunds sg,
         "staking" .= sgStaking sg
       ]
+  toEncoding sg =
+    -- Forced evaluation of each field allows the parent object to no longer be
+    -- referenced, which helps prevent retention of any fields that have already
+    -- been serialised.
+    let !systemStart = sgSystemStart sg
+        !networkMagic = sgNetworkMagic sg
+        !networkId = sgNetworkId sg
+        !activeSlotsCoeff = sgActiveSlotsCoeff sg
+        !securityParam = sgSecurityParam sg
+        !epochLength = sgEpochLength sg
+        !slotsPerKESPeriod = sgSlotsPerKESPeriod sg
+        !maxKESEvolutions = sgMaxKESEvolutions sg
+        !slotLength = sgSlotLength sg
+        !updateQuorum = sgUpdateQuorum sg
+        !maxLovelaceSupply = sgMaxLovelaceSupply sg
+        !protocolParams = sgProtocolParams sg
+        !genDelegs = sgGenDelegs sg
+        !initialFunds = sgInitialFunds sg
+        !staking = sgStaking sg
```

Comment:
If you want to ensure `sgInitialFunds` and `sgStaking` are strict then it would make sense to manually bind them with a `!`, but for other fields that are already strict this is just unnecessary boiler plate.  Can you please switch all others to `NamedFieldPuns`